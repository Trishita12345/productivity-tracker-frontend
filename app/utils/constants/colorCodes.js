export const COLORS = {
  LIGHT_GREY: '#F5F5F5',
  GREY: '#C6C6C6',
  DARK_GREY: '#87797F',
  VIOLET: '#70367C',
  ORANGE: '#FD7E50',
  SKY: '#95D0D5',
  DARK_VIOLET: '#301137',
  GREEN: '#6DB33F',
  YELLOW: '#DCD41D',
  RED: '#E41515',
  WHITE: '#FFFAFF',
  BLUE: '#4E63B2',
};
