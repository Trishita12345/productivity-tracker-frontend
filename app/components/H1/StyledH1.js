import styled from 'styled-components';
import { COLORS } from '../../utils/constants/colorCodes';

const textColor = {
  primary: COLORS.DARK_VIOLET,
  secondary: COLORS.WHITE,
  tertiary: COLORS.DARK_GREY,
};

const StyledH1 = styled.h1`
  font-family: 'Roboto Slab';
  font-style: normal;
  font-weight: bold;
  font-size: 2em;
  line-height: 24px;
  color: ${({ variant }) => textColor[variant]};
  margin-bottom: 0.25em;
`;

export default StyledH1;
