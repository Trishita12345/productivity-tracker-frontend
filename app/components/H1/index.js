import React from 'react';
import PropTypes from 'prop-types';

import StyledH1 from './StyledH1';
import { VARIANTS } from '../../utils/constants/variants';

const H1 = ({ label, variant = VARIANTS.PRIMARY }) => (
  <StyledH1 variant={variant}>{label}</StyledH1>
);

H1.propTypes = {
  label: PropTypes.string,
  variant: PropTypes.oneOf([
    VARIANTS.PRIMARY,
    VARIANTS.SECONDARY,
    VARIANTS.TERTIARY,
  ]),
};

export default H1;
