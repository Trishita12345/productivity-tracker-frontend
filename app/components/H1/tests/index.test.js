/**
 * Testing our H1 component
 */

import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import H1 from '../index';
import { VARIANTS } from '../../../utils/constants/variants';
import StyledH1 from '../StyledH1';

configure({ adapter: new Adapter() });

describe('<H1 />', () => {
  const props = {
    label: 'Label',
  };

  const ComponentRenderer = myProps => {
    const component = shallow(<H1 {...myProps} />);
    return component;
  };

  it('should H1 renders', () => {
    const component = ComponentRenderer(props);
    expect(component).toBeDefined();
    expect(component.find(StyledH1).text()).toEqual('Label');
  });
  it('should renders primary as default', () => {
    const component = ComponentRenderer(props);
    expect(component.find(StyledH1).prop('variant')).toEqual(VARIANTS.PRIMARY);
    expect(component).toMatchSnapshot();
  });
  it('should renders secondary variant', () => {
    const component = ComponentRenderer({
      ...props,
      variant: VARIANTS.SECONDARY,
    });
    expect(component.find(StyledH1).prop('variant')).toEqual(
      VARIANTS.SECONDARY,
    );
    expect(component).toMatchSnapshot();
  });
  it('should renders Tertiary variant', () => {
    const component = ComponentRenderer({
      ...props,
      variant: VARIANTS.TERTIARY,
    });
    expect(component.find(StyledH1).prop('variant')).toEqual(VARIANTS.TERTIARY);
    expect(component).toMatchSnapshot();
  });
});
