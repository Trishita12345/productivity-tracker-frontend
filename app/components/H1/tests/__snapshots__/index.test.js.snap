// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> should renders Tertiary variant 1`] = `ShallowWrapper {}`;

exports[`<H1 /> should renders primary as default 1`] = `ShallowWrapper {}`;

exports[`<H1 /> should renders secondary variant 1`] = `ShallowWrapper {}`;
